README.md

```diff
@@ -5,21 +5,97 @@ Right now this app only works on android. iOS support is planned in the future.
+
+### Node and JDK setup
 
 This app is a react-native based app with java bridges to interface between android core API's and the react-native component.
-Basic setup instructions for react-native projects along with dependencies can be found [here](https://reactnative.dev/docs/environment-setup?guide=native&platform=android)
-Please follow this guide exactly.
+Install node and watchman. If you have already installed Node on your system, make sure it is Node 18 or newer. Watchman is a tool by Facebook for watching changes in the filesystem. It is highly recommended you install it for better performance.
+
+```sh
+brew install node
+brew install watchman
+```
+We recommend installing the OpenJDK distribution called Azul Zulu.
+
+```sh
+brew tap homebrew/cask-versions
+brew install --cask zulu17
+
+# Get path to where cask was installed to double-click installer
+brew info --cask zulu17
+```
+
+After you install the JDK, update your JAVA_HOME environment variable. If you used above steps, JDK will likely be at `/Library/Java/JavaVirtualMachines/zulu-17.jdk/Contents/Home`
+
+If you have already installed JDK on your system, we recommend JDK 17. You may encounter problems using higher JDK versions. To make sure you are using the JDK version 17 update the `JAVA_HOME` and `PATH` variable to reflect the same.
+
+```sh
+export JAVA_HOME=/Library/Java/JavaVirtualMachines/zulu-17.jdk/Contents/Home/
+export PATH=$JAVA_HOME/bin:$PATH
+```
+
+### Setup Android Studio
+
+[Download and Install Android Studio](https://developer.android.com/studio/index.html)
+
+While on Android Studio installation wizard, make sure the boxes next to all of the following items are checked:
+
+- Android SDK
+- Android SDK Platform
+- Android Virtual Device
+
+Then, click "Next" to install all of these components.
+
+Android Studio installs the latest Android SDK by default. Building a React Native app with native code, however, requires the `Android 14 (UpsideDownCake)` SDK in particular. Additional Android SDKs can be installed through the SDK Manager in Android Studio.
+
+To do that, open Android Studio, click on "More Actions" button and select "SDK Manager".
 
-Create a `local.properties` file in `/android` folder and define the sdk path as shown below:
+Select the "SDK Platforms" tab from within the SDK Manager, then check the box next to "Show Package Details" in the bottom right corner. Look for and expand the `Android 14 (UpsideDownCake)` entry, then make sure the following items are checked:
 
+- Android SDK Platform
+- `Intel x86 Atom_64 System Image` or `Google APIs Intel x86 Atom System Image` or (for Apple M1 Silicon) `Google APIs ARM 64 v8a System Image`
+
+You also need to setup your environment variables to use the android utlities just installed. In your `~/.zshrc` or `~/.zprofile` (or `~/.bashrc` file if use bash shell) add the following lines
```

Comment:
In order to effortlessly use the previously installed android utilities, add the following lines in the `~/.zshrc` or `~/.zprofile` (or `~/.bashrc` file if using bash shell)